sklearn/feature_extraction/tests/test_feature_hasher.py

```diff
@@ -106,3 +106,44 @@ def test_hasher_zeros():
     # Assert that no zeros are materialized in the output.
     X = FeatureHasher().transform([{'foo': 0}])
     assert_equal(X.data.shape, (0,))
+
+
+def test_hasher_alternate_sign():
+    X = [["foo", "bar", "baz", "investigation need", "records"]]
+    # the last two tokens produce a hash collision that sums as 0
+
+    Xt = FeatureHasher(alternate_sign=True, non_negative=False,
+                       input_type='string').fit_transform(X)
+    assert_true(Xt.data.min() < 0 and Xt.data.max() > 0)
+    # check that we have a collision that produces a 0 count
+    assert_true(len(Xt.data) < len(X[0]))
+    assert_true((Xt.data == 0.).any())
+
+    Xt = FeatureHasher(alternate_sign=True, non_negative=True,
```

Comment:
This will raise a deprecation warning right? can you wrap it using `ignore_warnings(DeprecatedWarning)` maybe?